src/TestingUtils/Microsoft.AspNetCore.Testing/src/TestContext.cs

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Reflection;
+using Xunit.Abstractions;
+
+namespace Microsoft.AspNetCore.Testing
+{
+    /// <summary>
+    /// Provides access to contextual information about the running tests. Get access by
+    /// implementing <see cref="ITestMethodLifecycle"/>.
+    /// </summary>
+    /// <remarks>
+    /// Requires defining <see cref="AspNetTestFramework"/> as the test framework.
+    /// </remarks>
+    public sealed class TestContext
```

Comment:
How about Theory data?

Reply:
What do you mean? This context is set per-line-of-theory-data because that's xUnit's granularity. Each line-of-theory-data results in a new instance of the test class, etc.

Reply:
I mean the data for that instance of the theory.

Reply:
yes then.